docs/guides/writing-relationships.md

```diff
@@ -0,0 +1,88 @@
+# Writing Relationships
+
+:::note
+Want to learn more about writing relationships to SpiceDB, the various strategies and their pros and cons? Read our [blog post about writing relationships]
+:::
+
+[blog post about writing relationships]: https://authzed.com/blog/writing-relationships-to-spicedb/
+
+In [SpiceDB], a permissions system is defined by two items: the [schema] which defines *how* data can be represented, and the *relationships*, defining the way the objects are actually related to one another.
+
+[schema]: /guides/schema
+[SpiceDB]: https://github.com/authzed/spicedb
+
+It is the application's responsibility to keep the relationships within SpiceDB up-to-date and reflecting the state of the application; how an application does so can vary based on the specifics of the application, so below we outline a few approaches.
+
+## Two writes + commit
+
+The most common and straightforward way to store relationships in SpiceDB is to use a 2 phase commit-like approach, making use of a transaction from the relational database along with a [WriteRelationships] call to SpiceDB.
+
+[WriteRelationships]: https://buf.build/authzed/api/docs/main/authzed.api.v1#WriteRelationships
+
+```python title='Example of a 2PC-like approach'
+# Open the database transaction
+try:
+    with db.transaction() as transaction:
+        # Update the relationship in the database.
+        document = Document(
+            id="somedoc",
+            owner=some_user,
+        )
+        transaction.add(document)
+
+        # Add the relationship(s) in SpiceDB.
+        resp = client.WriteRelationships(...)
+
+        # Store the ZedToken we've received.
+        document.zedtoken = result.written_at
+
+        # Transaction is committed on close
+except:
+    # Delete the relationship(s) just written
+    client.DeleteRelationships(...)
+    raise
```

Comment:
I took a stab at making this more understandable. I'm using less python constructs so that it's explicit.

```py
try:
    tx = db.transaction()
    
    # Write relationships during a transaction so that it can be aborted on exception
    resp = spicedb_client.WriteRelationships(...)

    tx.add(db_models.Document(
        id=request.document_id,
        owner=user_id,
        zedtoken=resp.written_at
    ))
    tx.commit()
except:
    # Delete relationships written to SpiceDB and re-raise the exception
    tx.abort()
    spicedb_client.DeleteRelationships(...)
    raise
```